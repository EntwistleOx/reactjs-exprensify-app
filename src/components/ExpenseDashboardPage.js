import React from 'react'

const ExpenseDashboardPage = () => (
    <div>
        Dashboard component ok!
    </div>
)

export default ExpenseDashboardPage