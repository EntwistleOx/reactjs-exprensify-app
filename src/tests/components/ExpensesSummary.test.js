import React from 'react';
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import getExpensesTotal from '../../selectors/expenses-total'

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={14500} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multi expenses', () => {
    const total = getExpensesTotal(expenses)
    const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length} expensesTotal={total} />)
    expect(wrapper).toMatchSnapshot()
})
