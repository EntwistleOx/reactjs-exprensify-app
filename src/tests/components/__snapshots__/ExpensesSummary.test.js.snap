// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={14500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Viewing 
        1
         
        expense
         totalling 
        $145.00
      </h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$145.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totalling ",
        "$145.00",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Viewing 
          1
           
          expense
           totalling 
          $145.00
        </h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " ",
            "expense",
            " totalling ",
            "$145.00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$145.00",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`should render ExpensesSummary with expense msg 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={14500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Viewing 
        1
         
        expense
         totalling 
        $145.00
      </h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$145.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totalling ",
        "$145.00",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Viewing 
          1
           
          expense
           totalling 
          $145.00
        </h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " ",
            "expense",
            " totalling ",
            "$145.00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$145.00",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`should render ExpensesSummary with expenses msg 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={3}
    expensesTotal={114195}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Viewing 
        3
         
        expenses
         totalling 
        $1,141.95
      </h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          3,
          " ",
          "expenses",
          " totalling ",
          "$1,141.95",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        3,
        " ",
        "expenses",
        " totalling ",
        "$1,141.95",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Viewing 
          3
           
          expenses
           totalling 
          $1,141.95
        </h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            3,
            " ",
            "expenses",
            " totalling ",
            "$1,141.95",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          3,
          " ",
          "expenses",
          " totalling ",
          "$1,141.95",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`should render ExpensesSummary with multi expenses 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={3}
    expensesTotal={114195}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Viewing 
        3
         
        expenses
         totalling 
        $1,141.95
      </h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          3,
          " ",
          "expenses",
          " totalling ",
          "$1,141.95",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        3,
        " ",
        "expenses",
        " totalling ",
        "$1,141.95",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Viewing 
          3
           
          expenses
           totalling 
          $1,141.95
        </h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            3,
            " ",
            "expenses",
            " totalling ",
            "$1,141.95",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          3,
          " ",
          "expenses",
          " totalling ",
          "$1,141.95",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
